### PR TITLE
feat: Introduce a new CR for managing Notifications Configuration

### LINKED ISSUE #1265: Add a new custom resource to manage the argocd-notifications-cm
**Is your feature request related to a problem? Please describe.**
Currently, Argo CD Operator supports `Notifications` as Beta. One of the primary reasons for this is, Argo CD Operator does not manage the configuration(templates, triggers, services, subscriptions e.t.c.,.) of the the `argocd-notifications-cm`.

Triggers, Templates, Services and Subscriptions can be so huge at times that it creates a bad user experience if they are added to the Argo CD CR and managed by the Argo CD Reconciler of the Argo CD Operator. 

**Describe the solution you'd like**
- It is better to create a new CR - `NotificationsConfiguration` that can be used by the Users to update the their custom templates, triggers e.t.c.,. 
- A new reconciler will watch this CR to update the `argocd-notifications-cm`.
- `argocd-notifications-cm` should be created with default triggers, templates e.t.c.,. OOTB.

**Additional context**
Sample NotificationsConfiguration resource
```
apiVersion: argoproj.io/v1alpha1
kind: NotificationsConfiguration
metadata:
 name: default-notifications-configuration
spec:
 triggers:
  trigger.on-sync-status-unknown: |
    - when: app.status.sync.status == 'Unknown'
      send: [my-custom-template]
 templates:
  template.my-custom-template: |
    message: |
      Application details: {{.context.argocdUrl}}/applications/{{.app.metadata.name}}.
```

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind enhancement

**What does this PR do / why we need it**:
Fixes #1265 
Closes #1265 

**Have you updated the necessary documentation?**
* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:
Fixes #1265 

**How to test changes / Special notes to the reviewer**:
- Added unit tests to validate the behavior.
- Update/Delete the default(OOTB) NotificationsConfiguration CR to see if the changes are updated as expected in the `argocd-notifications-cm`

Note: This PR also updates docs to provide Instructions on how to update the `NotificationsConfiguration` CR.

### CODE REVIEW

File: controllers/argocd/notifications.go [hunk 1]
```diff
@@ -70,6 +72,28 @@ func (r *ReconcileArgoCD) reconcileNotificationsController(cr *argoproj.ArgoCD)
...
+			Name:      "default-notifications-configuration",
+			Namespace: cr.Namespace,
+		},
+	}
+
+	if err := argoutil.FetchObject(r.Client, cr.Namespace, "default-notifications-configuration",
+		defaultNotificationsConfigurationCR); err != nil {
+
+		if errors.IsNotFound(err) {
+			err := r.Client.Create(context.TODO(), defaultNotificationsConfigurationCR)
```
[the remark this answers is not in the document]
Comment (iam-veeramalla) [reply controllers/argocd/notifications.go]: Idea is to abstract them to keep the CR light. We will mention the list of default templates, triggers e.t.c.,. in the documentation.

[the remark this answers is not in the document]
Comment (iam-veeramalla) [reply controllers/argocd/notifications.go]: I gave it a second thought and it makes sense. Made the changes as suggested.

File: docs/reference/notificationsconfiguration.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# NotificationsConfiguration
+
+The `NotificationsConfiguration` resource is a Kubernetes Custom Resource (CRD) that allows users to add the triggers, templates, services and subscriptions to the Argo CD Notifications Configmap.
+
+A `NotificationsConfiguration` custom resource with name `default-notifications-configuration` is created **OOTB** with default configuration. Users should update this custom resource with their templates, triggers, services, subscriptios or any other configuration.
```
Comment (svghadi) [inline docs/reference/notificationsconfiguration.md:5]: Should we add an explicit important note here, stating that only updates to `default-notifications-configuration` are supported, and users cannot create new NotificationsConfiguration custom resources as of now?

**Review by svghadi — APPROVED** [review]
Tested out the changes. Looks good. Left a doc suggestion. Thanks!!

**Pull request merged**